### PR TITLE
Fix bin with 2d input edges

### LINKED ISSUE #1821: `bin` sometimes throws if 2-D coord is to be rebinned
I believe we have some passing unit tests for this case, but in practice it seems that something is not working, in particular for neutron event data obtained after converting units, e.g., to wavelength, then using `sc.bin` to change wavelength binning throws.

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #1821 by remove overly restrictive dim check in `accumulate_in_place`.

### CODE REVIEW

File: dataset/test/bin_test.cpp [hunk 1]
```diff
@@ -219,7 +219,7 @@ TEST_P(BinTest, rebin_2d_with_2d_coord) {
   auto table = GetParam();
   auto xy = bin(table, {edges_x_coarse, edges_y_coarse});
   Variable edges_y_2d = makeVariable<double>(Dims{Dim::X, Dim::Y}, Shape{2, 3},
-                                             Values{-2, 1, 2, -1, 0, 3});
+                                             Values{-2, 1, 2, -3, 0, 3});
```
Comment (jl-wynen) [inline dataset/test/bin_test.cpp:222]: Does this change mean anything?

**Review by jl-wynen — COMMENTED** [review]
It would be nice to test that the output variable is unchanged when the dimension check fails since this now happens later.  (It looks fine but doesn't hurt to make sure.)
Looks good otherwise.

File: dataset/test/bin_test.cpp (hunk 1 above)
Comment (SimonHeybrock) [reply dataset/test/bin_test.cpp:222]: Yes and now: This test uses a hacky way of adding 2-D edges, which resulted in events being outside the edges. Depending on whether also the out dim was rebinned, `bin` would or would not detect that and drop those events.

Comment (jl-wynen) [reply dataset/test/bin_test.cpp:222]: Gotcha

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**